### DISCUSSION TITLE
kubernetes service in default namespace

### DISCUSSION THREAD

**Discussion by asharma1988** [k3s-io/k3s#9830] [General]
Hi All,
Probably a dumb question but I am running k3s 1.27.4 and I have 5 nodes with all running k3s in server mode. In k3s service file '--service-cidr=172.16.4.0/22' is added, and in the cluster I have kubernetes service in default namespace with 172.16.4.1:443 with endpoints are k3s servers listening on 6443 on all 5 nodes. In the kubeconfig file I have a separate url for cluster.server, which is same as K3S_URL on the nodes. I am wondering how k3s creates kubernetes service with that service IP along with the endpoints as I am not configuring 172.16.4.1:443 anywhere. Thanks in advance.

**Comment by brandond** [comment]
The apiserver handles that internally. The kubernetes.default service is always available at the first address in the service-cidr block, and has endpoints for each of the active apiserver nodes.
